offchain-modules/src/packages/handlers/tron.ts

```diff
@@ -0,0 +1,270 @@
+import { TronDb } from '../db';
+import { logger } from '../utils/logger';
+import { asyncSleep } from '../utils';
+import { ForceBridgeCore } from '../core';
+import { TronWeb } from 'tronweb';
+import { TronGrid } from 'trongrid';
+import { TronLock, TronUnlock, ICkbMint } from '@force-bridge/db/model';
+import { ChainType } from '@force-bridge/ckb/model/asset';
+
+type TronLockEvent = {
+  tx_hash: string;
+  index?: number;
+  sender: string;
+  asset: string;
+  amount: string;
+  memo: string;
+  timestamp: number;
+};
+
+export class TronHandler {
+  private tronWeb: TronWeb;
+  private tronGrid: TronGrid;
+  private committee;
+  constructor(private db: TronDb) {
+    this.tronWeb = new TronWeb({ fullHost: ForceBridgeCore.config.tron.tronGridUrl });
+    this.tronGrid = new TronGrid(this.tronWeb);
+    this.committee = ForceBridgeCore.config.tron.committee;
+  }
+
+  private async getTrxAndTrc10LockEvents(min_timestamp: number): Promise<TronLockEvent[]> {
+    const txs = await this.tronGrid.account.getTransactions(this.committee.address, {
+      only_to: true,
+      only_confirmed: true,
+      min_timestamp: min_timestamp,
+    });
+
+    let lockEvents: TronLockEvent[];
+    for (let i = 0; i < txs.data.length; i++) {
+      const asset_data = txs.data[i].raw_data.contract[0].parameter.value;
+      lockEvents[i] = {
+        tx_hash: txs.data[i].txID,
+        sender: this.tronWeb.address.fromHex(asset_data.owner_address),
+        asset: asset_data.asset_name ? asset_data.asset_name : 'trx',
+        amount: asset_data.amount,
+        memo: this.tronWeb.toUtf8(txs.data[i].raw_data.data),
+        timestamp: txs.data[i].block_timestamp,
+      };
+    }
+    return lockEvents;
+  }
+
+  private async getTrc20TxsLockEvents(min_timestamp: number): Promise<TronLockEvent[]> {
+    const txs = await this.tronGrid.account.getTrc20Transactions(this.committee.address, {
+      only_confirmed: true,
+      only_to: true,
+      min_timestamp: min_timestamp,
+    });
+
+    let lockEvents: TronLockEvent[];
+    for (let i = 0; i < txs.data.length; i++) {
+      const tx = await this.tronWeb.trx.getTransaction(txs.data[i].transaction_id);
+      lockEvents[i] = {
+        tx_hash: txs.data[i].transaction_id,
+        sender: txs.data[i].from,
+        asset: this.tronWeb.address.fromHex(txs.data[i].token_info).address,
+        amount: txs.data[i].value,
+        memo: this.tronWeb.toUtf8(tx.raw_data.data),
+        timestamp: txs.data[i].block_timestamp,
+      };
+    }
+    return lockEvents;
+  }
+
+  // memo style should be "ckb_recipient,sudt_extra_data"
+  private analyzeMemo(memo: string) {
+    const splitted = memo.split(',', 2);
+    const ckbRecipient = splitted[0];
+    const sudtExtraData = splitted[1];
```

Comment:
should handle error here to avoid exception.

Reply:
if string not contains ','， split will return undefined rather error